package-res/resources/web/pentaho/type/simple.js

```diff
@@ -149,11 +175,42 @@ define([
         return f != null ? f : String(this._value);
       },
 
+      /**
+       * Gets the key of the simple value.
+       *
+       * The key of a value identifies it among values of the same concrete type.
+       *
+       * If two values have the same concrete type and their
+       * keys are equal, then it must also be the case that
+       * {@link pentaho.type.Value.Meta#areEqual}
+       * returns `true` when given the two values.
+       * The opposite should be true as well.
+       * If two values of the same concrete type have distinct keys,
+       * then {@link pentaho.type.Value.Meta#areEqual} should return `false`.
+       *
+       * The default simple value implementation, returns the result of calling
+       * `toString()` on {@link pentaho.type.Simple#value}.
+       *
+       * @type string
+       * @readonly
+       */
+      get key() {
```

Comment:
This don't work as intended with ```pentaho/type/object```. The ```toString``` always return ```"[object Object]"```, so it is always considered equal.

Maybe in this case we could probably just consider equals if it indeed is the same instance. That would need an override of the ```equals``` function by ```pentaho/type/object``` so that don't check the ```key``` property. The doc description of this getter would also have to change to break the hard requirement between the two functions.

Other option is to override the ```key``` getter in ```pentaho/type/object``` so it returns an hash of the object (i.e., ```md5``` of a ```JSON.stringify```). It would become, however, a much costly function.